prime-router/src/test/kotlin/fhirengine/translation/hl7/FhirTransformerTests.kt

```diff
@@ -1240,6 +1242,92 @@ class FhirTransformerTests {
         assertThat(patient.name).hasSize(2)
     }
 
+    @Test
+    fun `test accessing by index while setting the actual bundle property`() {
+        val bundle = Bundle()
+        bundle.id = "abc123"
+        val patient = Patient()
+        val name = HumanName()
+        name.given = mutableListOf(StringType("foo"), StringType("bar"))
+        patient.name = mutableListOf(name)
+        patient.id = "def456"
+        val patientEntry = bundle.addEntry()
+        patientEntry.fullUrl = patient.id
+        patientEntry.resource = patient
+
+        val updateFirstGivenNameSchemaElement = FhirTransformSchemaElement(
+            "update-first-given-name",
+            resource = "Bundle.entry.resource.ofType(Patient).name",
+            bundleProperty = "%resource.given[0]",
+            value = listOf("''")
+        )
+        val schema = FhirTransformSchema(elements = mutableListOf(updateFirstGivenNameSchemaElement))
+
+        val transformer = FhirTransformer(schema)
+        val exception = assertThrows<ConfigSchemaElementProcessingException> {
+            transformer.process(bundle)
+        }
+        assertThat(exception.message)
+            .contains("Schema is attempting to set a value for a particular index which is not allowed")
+    }
+
+    @Test
+    fun `test deidentify human name`() {
+        val bundle = Bundle()
+        bundle.id = "abc123"
+        val patient = Patient()
+        val name = HumanName()
+        name.given = mutableListOf(StringType("foo"), StringType("bar"))
+        name.family = "family"
+        patient.name = mutableListOf(name)
+        patient.id = "def456"
+        val patientEntry = bundle.addEntry()
+        patientEntry.fullUrl = patient.id
+        patientEntry.resource = patient
+
+        val updateFirstGivenNameSchemaElement = FhirTransformSchemaElement(
+            "update-first-given-name",
+            resource = "Bundle.entry.resource.ofType(Patient)",
+            bundleProperty = "%resource.name",
+            function = "deidentifyHumanName()",
+        )
+        val schema = FhirTransformSchema(elements = mutableListOf(updateFirstGivenNameSchemaElement))
+
+        val transformer = FhirTransformer(schema)
+        transformer.process(bundle)
+        assertThat(name.given).isEmpty()
```

Comment:
I can't remember from your demo the other day, does this work when there are multiple Patient.name results?

Reply:
yep, it will because we're selecting all `Patient` resources